plutus-scb/src/Plutus/SCB/Effects/EventLog.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE DataKinds        #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs            #-}
+{-# LANGUAGE LambdaCase       #-}
+{-# LANGUAGE TemplateHaskell  #-}
+{-# LANGUAGE TypeOperators    #-}
+
+module Plutus.SCB.Effects.EventLog where
+
+import           Control.Lens
+import           Control.Monad.Freer
+import           Control.Monad.Freer.Extras (monadStateToState)
+import           Control.Monad.Freer.Reader
+import           Control.Monad.Freer.State  (State)
+import           Control.Monad.Freer.TH     (makeEffect)
+import qualified Control.Monad.IO.Unlift    as Unlift
+import qualified Control.Monad.Logger       as MonadLogger
+import           Data.Aeson                 (FromJSON, ToJSON)
+import           Database.Persist.Sqlite    hiding (Connection)
+import           Eventful
+import qualified Eventful.Store.Memory      as M
+import           Eventful.Store.Sql
+import           Eventful.Store.Sqlite      (sqliteEventStoreWriter)
+
+import           Plutus.SCB.Events          (ChainEvent (..))
+import           Plutus.SCB.Query           (nullProjection)
+import           Plutus.SCB.Types           (Source (..), toUUID)
+
+newtype Connection =
+    Connection (SqlEventStoreConfig SqlEvent JSONString, ConnectionPool)
+
+
+-- | Event effects
+data EventLogState pjs =
+     EventLogState { _elsEvents      :: [ChainEvent]
+                   , _elsProjections :: pjs }
+makeLenses ''EventLogState
+
+data EventLogEffect event r where
+    RefreshProjection :: GlobalStreamProjection state event
+                     -> EventLogEffect event (GlobalStreamProjection state event)
+    RunCommand :: Aggregate state event command -> Source -> command -> EventLogEffect event [event]
+makeEffect ''EventLogEffect
+
+handleEventLogState ::
```

Comment:
Should probably be `handleEventLogEffect`.

Reply:
I wanted to capture the fact that it uses the `State` effect - I changed the name of the other handler to `handleEventLogSql` to make that clearer.